x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -393,6 +405,33 @@ export class TaskManagerRunner implements TaskRunner {
     return { ...(error ? { error } : {}), state };
   }
 
+  private validateTaskState(taskInstance: ConcreteTaskInstance) {
+    let error;
+    const { taskType, id, numSkippedRuns = 0 } = taskInstance;
+    const { max_attempts: maxAttempts } = this.requeueInvalidTasksConfig;
+    let validatedTask;
+    try {
+      validatedTask = this.taskValidator.getValidatedTaskInstanceFromReading(taskInstance);
+      return {
+        validatedTaskInstance: validatedTask,
+        validatedState: { state: validatedTask.state },
+      };
+    } catch (err) {
+      this.logger.warn(`Task (${taskType}/${id}) has a validation error: ${err.message}`);
+      if (numSkippedRuns < maxAttempts) {
```

Comment:
It doesn't look like `numSkippedRuns` is incremented if task state validation fails, so the task will continue getting claimed indefinitely. Is that the behavior we want or do we want it to stop trying after X number of attempts? If we do want to continue trying indefinitely, should we remove this logic for checking `numSkippedRuns`?

Reply:
Thank ying! I am not totally sure what behavior we want, @mikecote what do you think here?

Reply:
Should be resolved in this commit [5362f27](https://github.com/elastic/kibana/pull/170808/commits/5362f27a780ed97ffc238c3a0df7a4963f5f2fb4)

The validation will throw an error that will cause the task to fail.